tests/eval/test_in_context_learning_datasets.py

```diff
@@ -0,0 +1,2880 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import contextlib
+import os
+import random
+import types
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import pytest
+import torch
+from composer import Evaluator
+from composer.core import DataSpec
+from torch.utils.data import DataLoader
+
+# isort: off
+from llmfoundry.eval.datasets import (
+    InContextLearningDataset, InContextLearningCodeEvalDataset,
+    InContextLearningMultipleChoiceTaskDataset,
+    InContextLearningGenerationTaskWithAnswersDataset,
+    InContextLearningSchemaTaskDataset, get_icl_task_dataloader, strip_data,
+    tokenizer_needs_prefix_space, trim_context, get_continuation_span,
+    get_fewshot_sample_idxs, make_padded_input)
+# isort: on
+import transformers
+from composer.datasets.utils import MultiTokenEOSCriteria
+from composer.loggers import InMemoryLogger
+from composer.models import HuggingFaceModel
+from composer.trainer import Trainer
+from composer.utils import dist, reproducibility
+
+from llmfoundry.eval.metrics import (
+    InContextLearningCodeEvalAccuracy,
+    InContextLearningGenerationExactMatchAccuracy, InContextLearningLMAccuracy,
+    InContextLearningMultipleChoiceAccuracy)
+
+
+def test_strip_data():
+    data_to_strip = {
+        'strip_data': '  boo!  \n',
+        'has_space': '  wa hoo!',
+        'end_space': 'yoohoo!  '
+    }
+    stripped_data = strip_data(data_to_strip)
+    for k, v in stripped_data.items():
+        assert k in data_to_strip
+        assert not v[0].isspace()
+        assert not v[-1].isspace()
+
+
+@pytest.mark.skip(
+    reason="Currently don't have a tokenizer that satisfies this test")
+def test_tokenizer_needs_prefix_space_when_space_not_needed(
+        tiny_gpt2_tokenizer: transformers.AutoTokenizer):
+    assert not tokenizer_needs_prefix_space(tiny_gpt2_tokenizer)
+
+
+def test_tokenizer_needs_prefix_space_when_space_needed():
+    transformers = pytest.importorskip('transformers')
```

Comment:
you can get rid of all of the importorskip of transformers and datasets just import at the top now